File: src/show_me_tokens/adapters/registry.py
"""Adapter registry."""

from __future__ import annotations

from show_me_tokens.adapters.base import AgentAdapter
from show_me_tokens.adapters.qoder import QoderAdapter
from show_me_tokens.adapters.qoderwork import QoderWorkAdapter

ADAPTERS: dict[str, type[AgentAdapter]] = {
    QoderAdapter.name: QoderAdapter,
    QoderWorkAdapter.name: QoderWorkAdapter,
}


def get_adapter(name: str) -> type[AgentAdapter]:
    """Return the adapter class registered under ``name``.

    Raises:
        KeyError: if no adapter is registered for ``name``.
    """
    try:
        return ADAPTERS[name]
    except KeyError as exc:
        raise KeyError(
            f"Unknown agent: {name!r}. "
            f"Supported agents: {', '.join(sorted(ADAPTERS))}"
        ) from exc


def list_adapter_names() -> list[str]:
    """Return supported adapter names in alphabetical order."""
    return sorted(ADAPTERS)

File: src/adapters/registry.ts
import { AgentAdapter } from './base.js';
import { QoderAdapter } from './qoder.js';
import { QoderWorkAdapter } from './qoderwork.js';

export type AdapterConstructor = new () => AgentAdapter;

const ADAPTERS: Record<string, AdapterConstructor> = {
  qoder: QoderAdapter,
  qoderwork: QoderWorkAdapter,
};

export function getAdapter(name: string): AdapterConstructor {
  const cls = ADAPTERS[name];
  if (!cls) {
    throw new Error(
      `Unknown agent: '${name}'. Supported agents: ${listAdapterNames().join(', ')}`
    );
  }
  return cls;
}

export function listAdapterNames(): string[] {
  return Object.keys(ADAPTERS).sort();
}
